### PR TITLE
Fixed#9427: LDAP-s support connection without MTLS

### LINKED ISSUE #9427: LDAP-s support connection without MTLS
Support secure LDAP connection without MTLS (verification of server) requirement

A LDAPs (secure LDAP connection) should not require a client certificate.

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Fixes #9427 

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@mohityadav766 @harshach

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java [hunk 1]
```diff
@@ -77,26 +84,52 @@ public void init(OpenMetadataApplicationConfig config, Jdbi jdbi) {
...
+        X509TrustManager x509TrustManager;
+        SSLSocketVerifier sslSocketVerifier;
         LDAPConnectionOptions connectionOptions = new LDAPConnectionOptions();
-        SSLSocketVerifier sslSocketVerifier =
-            ldapConfiguration.getVerifyCertificateHostname()
-                ? new HostNameSSLSocketVerifier(true)
-                : TrustAllSSLSocketVerifier.getInstance();
-        connectionOptions.setSSLSocketVerifier(sslSocketVerifier);
+        LdapConfiguration.TruststoreConfigType configType = ldapConfiguration.getTruststoreConfigType();
+        switch (configType) {
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java:98]: move this to LdapUtil

File: openmetadata-docs/content/deployment/security/ldap/index.md [hunk 2]
```diff
@@ -66,11 +65,22 @@ Please see the below image for a sample LDAP Configuration in ApacheDS.
...
 
 - `maxPoolSize`: Connection Pool Size to use to connect to LDAP Server.
-- `sslEnabled`: Set to true if the SSL is enable to connecto to LDAP Server.
-- `keyStorePath`: Path of Keystore in case the sslEnabled is set to true.
-- `keyStorePassword`: Truststore Password.
-- `truststoreFormat`: TrustStore Format (Example :- JKS).
-- `verifyCertificateHostname`: Controls using TrustAllSSLSocketVerifier vs HostNameSSLSocketVerifier. In case the certificate contains cn=hostname of the Ldap Server set it to true.
+- `sslEnabled`: Set to true if the SSL is enable to connect to LDAP Server.
+- `truststoreConfigType`: Truststore type. Can select from {CustomTrustStore, HostName, JVMDefault, TrustAll}
+- `trustStoreConfig`: Config for the selected truststore type.
```
Comment (mohityadav766) [inline openmetadata-docs/content/deployment/security/ldap/index.md:70]: please mention one of these is required , and the config will differ based on that

File: openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java (hunk 1 above)
Comment (parthp2107) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java]: addressed in the latest commit.

File: openmetadata-docs/content/deployment/security/ldap/index.md (hunk 2 above)
Comment (parthp2107) [reply openmetadata-docs/content/deployment/security/ldap/index.md]: addressed in the latest commit

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**